python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -419,10 +418,98 @@ def deduplicates(self):
+            import pandas as pd
+            df = self.to_spark_df()
+            sort_df = df.sort(col_names, ascending=ascending)
+            sort_df = sort_df.coalesce(1)
+            data_shards = spark_df_to_pd_sparkxshards(sort_df)
+            return data_shards
+        else:
+            invalidInputError(False,
+                              "Currently only support sort() on XShards of Pandas DataFrame")
+
+    def max_values(self, col_name):
+        """
+        Get the max values of the column name. This is only applicable for SparkXShards
+         of Pandas Series.
+
+        :param col_name column name that need return the max value
+        :return: maximum for the specified columns.
+        """
+        if self._get_class_name() == 'pandas.core.frame.DataFrame':
+            import pandas as pd
+            df = self.to_spark_df()
+            from pyspark.sql.functions import max
+            max_value = df.select(max(col_name)).collect()
+            return max_value
+        else:
+            invalidInputError(False,
+                              "Currently only support max() on XShards of Pandas DataFrame")
+
+    def get_null_sum(self):
+        """
+        With SparkXShards of pandas data frame, the api will get null numbers for
+        each column. For other type of SparkXShards, it will throw exception
+
+       :return: SparkXShards of pandas data frame with 2 columns, `col` represents column name,
+        `total` represents null numbers
+        """
+        if self._get_class_name() != 'pandas.core.frame.DataFrame':
+            invalidInputError(False,
+                              "Currently only support assembleFeatureLabelCols() on"
+                              " XShards of Pandas DataFrame")
+
+        def get_na_sum(df):
+            import pandas as pd
+            series = df.isnull().sum()
+            df = pd.DataFrame({'col': series.index, 'total': series.values})
+            return df
+
+        na_cnt_shard = self.transform_shard(get_na_sum)
+
+        from functools import reduce
+        sum_rdd = na_cnt_shard.rdd.mapPartitions(
+            lambda iter: [reduce(lambda l1, l2: l1.add(l2), iter)])
+        sum_shards = SparkXShards(sum_rdd)
+        return sum_shards
+
+    def drop_missing_value(self):
+        """
+        With SparkXShards of pandas data frame, the api will drop null values in shards.
+         For other type of SparkXShards, it will throw exception
+
+       :return: a new SparkXShards without null values
+        """
+        if self._get_class_name() != 'pandas.core.frame.DataFrame':
+            invalidInputError(False,
+                              "Currently only support assembleFeatureLabelCols() on"
+                              " XShards of Pandas DataFrame")
+
+        null_cnt_shard = self.get_null_sum()
+        missing_data_shards = null_cnt_shard.sort_values(col_names="total", ascending=False)
+        zip_shards = self.zip(missing_data_shards)
```

Comment:
can you zip here? after sort, the new partitions do not match the original ones?

Reply:
Updated the code to ensure the new partitions matches the original one.